File: src/Entity/PollOption.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\PollOptionRepository;

#[ORM\Entity(repositoryClass: PollOptionRepository::class)]
class PollOption
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private $id;

    #[ORM\ManyToOne(targetEntity: Poll::class, inversedBy: 'options')]
    #[ORM\JoinColumn(nullable: false)]
    private $poll;

    #[ORM\Column(type: 'string', length: 255)]
    private $text;

    #[ORM\Column(type: 'integer')]
    private $voteCount = 0;

    #[ORM\OneToMany(targetEntity: Vote::class, mappedBy: 'option')]
    private $votes;

    public function __construct()
    {
        $this->votes = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getPoll(): ?Poll
    {
        return $this->poll;
    }

    public function setPoll(?Poll $poll): self
    {
        $this->poll = $poll;
        return $this;
    }

    public function getText(): ?string
    {
        return $this->text;
    }

    public function setText(string $text): self
    {
        $this->text = $text;
        return $this;
    }

    public function getVoteCount(): ?int
    {
        return $this->voteCount;
    }

    public function setVoteCount(int $voteCount): self
    {
        $this->voteCount = $voteCount;
        return $this;
    }

    public function incrementVoteCount(): self
    {
        $this->voteCount++;
        return $this;
    }

    public function decrementVoteCount(): self
    {
        $this->voteCount--;
        return $this;
    }

    /**
     * @return Collection|Vote[]
     */
    public function getVotes(): Collection
    {
        return $this->votes;
    }

    public function addVote(Vote $vote): self
    {
        if (!$this->votes->contains($vote)) {
            $this->votes[] = $vote;
            $vote->setOption($this);
        }
        return $this;
    }

    public function removeVote(Vote $vote): self
    {
        if ($this->votes->contains($vote)) {
            $this->votes->removeElement($vote);
            if ($vote->getOption() === $this) {
                $vote->setOption(null);
            }
        }
        return $this;
    }
} 

File: OneDrive/Bureau/Hackify-integrationn/src/main/java/models/PollOption.java
package models;

public class PollOption {
    private int id;
    private int pollId; // Foreign key to poll
    private String text; // Option text
    private int voteCount; // Number of votes

    // Default constructor
    public PollOption() {
    }

    // Constructor for creating a new option
    public PollOption(int pollId, String text) {
        this.pollId = pollId;
        this.text = text;
        this.voteCount = 0;
    }

    // Constructor for retrieving from database
    public PollOption(int id, int pollId, String text, int voteCount) {
        this.id = id;
        this.pollId = pollId;
        this.text = text;
        this.voteCount = voteCount;
    }

    // Getters and setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPollId() {
        return pollId;
    }

    public void setPollId(int pollId) {
        this.pollId = pollId;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public int getVoteCount() {
        return voteCount;
    }

    public void setVoteCount(int voteCount) {
        this.voteCount = voteCount;
    }
}